src/System.Management.Automation/engine/SessionStateLocationAPIs.cs

```diff
@@ -230,10 +231,31 @@ internal PathInfo SetLocation(string path, CmdletProviderContext context)
             ProviderInfo provider = null;
             string providerId = null;
 
+            // Replace path with last working directory when '-' was passed.
+            bool pushNextLocation = true;
+            if (originalPath.Equals("-", StringComparison.OrdinalIgnoreCase))
+            {
+                if (_workingLocationHistoryStack.Count > 0)
```

Comment:
I would structure this `if` as:
```csharp
if (_workingLocationHistoryStack.Count <= 0)
{
    throw new InvalidOperationException(...);
}

// Golden path behaviour
```

Just because there are some pretty deep `if`/`else` trees in the codebase and they get hard to reason about quickly.

Reply:
I agree, this is neater.